temp/OnnxTransformTempSolution/Microsoft.ML.Transforms.Onnx/OnnxTransform.cs

```diff
@@ -0,0 +1,294 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Microsoft.ML.Runtime;
+using Microsoft.ML.Runtime.CommandLine;
+using Microsoft.ML.Runtime.Data;
+using Microsoft.ML.Runtime.EntryPoints;
+using Microsoft.ML.Runtime.Internal.Utilities;
+using Microsoft.ML.Runtime.Model;
+using Microsoft.ML.Scoring;
+using Microsoft.ML.Transforms.Onnx;
+using OnnxShape = System.Collections.Generic.List<long>;
+
+[assembly: LoadableClass(OnnxTransform.Summary, typeof(IDataTransform), typeof(OnnxTransform.Arguments), typeof(SignatureDataTransform),
+    OnnxTransform.UserName, OnnxTransform.ShortName)]
+
+// This is for de-serialization from a binary model file.
+[assembly: LoadableClass(typeof(OnnxTransform.OnnxMapper), null, typeof(SignatureLoadRowMapper),
+    "", OnnxTransform.OnnxMapper.LoaderSignature)]
+
+namespace Microsoft.ML.Transforms.Onnx
+{
+    public static class OnnxTransform
+    {
+        public sealed class Arguments : TransformInputBase
+        {
+
+            [Argument(ArgumentType.Required, HelpText = "Path to the onnx model file.", ShortName = "model", SortOrder = 0)]
+            public string ModelFile;
+
+            [Argument(ArgumentType.Multiple | ArgumentType.Required, HelpText = "TBD", SortOrder = 1)]
+            public string InputColumn;
+
+            [Argument(ArgumentType.Multiple | ArgumentType.Required, HelpText = "TBD", SortOrder = 2)]
+            public string OutputColumn;
+
+            public OnnxModelInfo ModelInfo;
```

Comment:
>ModelInfo [](start = 33, length = 9)

I'm not sure I understand. Where does this get set?
Wouldn't it make more sense to have a static method that creates one of these given an Arguments object?

Reply:
OnnxModelInfo hold all the args that should be inferred from the model but are currently provided by user. This way it's easy to take it out once the sonoma API is ready. The creator of OnnxTransform should create it and pass it in. But it would be unnecessary later on.